### PR TITLE
update plugins section

### PR DESCRIPTION
**Pull request by silesky**
### Proposed changes

- explain difference between addSourceMiddleware and Plugins, and prefer addSourceMiddleware.
- correct description of plugins that suggest parallel vs synchronous loading -- we wait for all plugin types to io initialize.

### CODE REVIEW

**Comment by stayseesong** [comment]
@silesky should this be converted to draft mode since it's still wip?

**Comment by silesky** [comment]
> @silesky should this be converted to draft mode since it's still wip?

What makes you think it's still wip? The commit messages?

**Comment by stayseesong** [comment]
yeah

**Comment by silesky** [comment]
@stayseesong OK! those messages get squashed on merge, but I can squash them in advance.

File: src/connections/sources/catalog/libraries/website/javascript/index.md [hunk 1]
```diff
@@ -651,32 +651,68 @@ No, there is no change in behavior to Middlewares.
...
+Plugins and Source Middleware accomplish the same thing, but Plugins are significantly more powerful (but more verbose to implement).
+
+For basic use cases like adding event fields or dropping specific events, use [Source Middleware](#source-middleware). If you need more granular control of the lifecycle, or want to be able to abort segment initialization, you should reach for [Plugins](#plugins-for-advanced-use-cases).
+
+### Source Middleware 
+
+[Source middleware](docs/connections/sources/catalog/libraries/website/javascript/middleware/) runs before any other plugins. You can use this to enrich or drop an event.
+
+#### Example usage of `addSourceMiddleware`
+ d
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/website/javascript/index.md:669]:
```suggestion
```

File: src/connections/sources/catalog/libraries/website/javascript/index.md [hunk 2]
```diff
@@ -651,32 +651,68 @@ No, there is no change in behavior to Middlewares.
 #### When using Segment features (Schema filtering, integrations object, Protocols) to filter events from going to destinations (device and cloud-mode), will batching impact the filtering of events?
 No, there is no impact to how events filter.
 
-## Plugin architecture
-When you develop against Analytics 2.0, the plugins you write can augment functionality, enrich data, and control the flow and delivery of events. From modifying event payloads to changing analytics functionality, plugins help to speed up the process of getting things done.
+## Plugins and Source Middleware
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/website/javascript/index.md:654]:
```suggestion
## Plugins and source middleware
```

File: src/connections/sources/catalog/libraries/website/javascript/index.md [hunk 3]
```diff
@@ -651,32 +651,68 @@ No, there is no change in behavior to Middlewares.
...
 
-Though middlewares function the same as plugins, it's best to use plugins as they are easier to implement and are more testable.
+When you develop against Analytics 2.0, the plugins you write can augment functionality, enrich data, and control the flow and delivery of events. From modifying event payloads to changing analytics functionality, plugins and middleware help to speed up the process of getting things done.
 
+### What is the difference between Source Middleware and Plugins, and which should I use?
+
+Plugins and Source Middleware accomplish the same thing, but Plugins are significantly more powerful (but more verbose to implement).
+
+For basic use cases like adding event fields or dropping specific events, use [Source Middleware](#source-middleware). If you need more granular control of the lifecycle, or want to be able to abort segment initialization, you should reach for [Plugins](#plugins-for-advanced-use-cases).
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/website/javascript/index.md:662]:
```suggestion
Plugins and source middleware accomplish the same thing, but plugins are significantly more powerful (but more verbose to implement).

For basic use cases like adding event fields or dropping specific events, use [source middleware](#source-middleware). If you need more granular control of the lifecycle, or want to be able to abort the Segment initialization, you should use [plugins](#plugins-for-advanced-use-cases).
```

File: src/connections/sources/catalog/libraries/website/javascript/index.md [hunk 4]
```diff
@@ -651,32 +651,68 @@ No, there is no change in behavior to Middlewares.
...
+
+### Source Middleware 
+
+[Source middleware](docs/connections/sources/catalog/libraries/website/javascript/middleware/) runs before any other plugins. You can use this to enrich or drop an event.
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/website/javascript/index.md:666]:
```suggestion
[Source middleware](/docs/connections/sources/catalog/libraries/website/javascript/middleware/) runs before any other plugins. You can use this to enrich or drop an event.
```

File: src/connections/sources/catalog/libraries/website/javascript/index.md (hunk 1 above)
Comment (stayseesong) [reply src/connections/sources/catalog/libraries/website/javascript/index.md:669]: should there be something here?

File: src/connections/sources/catalog/libraries/website/javascript/index.md [hunk 5]
```diff
@@ -651,32 +651,68 @@ No, there is no change in behavior to Middlewares.
...
+
+#### Example usage of `addSourceMiddleware`
+ d
+#### Enrichment
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/website/javascript/index.md:670]: does this section just contain an example of enrichment?

File: src/connections/sources/catalog/libraries/website/javascript/index.md [hunk 6]
```diff
@@ -651,32 +651,68 @@ No, there is no change in behavior to Middlewares.
...
+analytics.addSourceMiddleware(({ payload, next }) => {
+   const { event } = payload.obj.context
+   if (event.type === 'track') {
+      event.event.toLowerCase()
+   }
+   next(payload)
+});
+```
+
+#### Validation
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/website/javascript/index.md:681]: does this section contain an example of validation?

File: src/connections/sources/catalog/libraries/website/javascript/index.md [hunk 7]
```diff
@@ -651,32 +651,68 @@ No, there is no change in behavior to Middlewares.
...
+analytics.addSourceMiddleware(({ payload, next }) => {
+  const { event } = payload.obj.context
+  if (!isValid(event)) {
+    throw new Error("Event will be dropped")
+  }
+  next(payload)
+});
+```
+
+### Plugins (For advanced use-cases)
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/website/javascript/index.md:692]: Is there supposed to be something here?

**Review by stayseesong — CHANGES_REQUESTED** [review]

File: src/connections/sources/catalog/libraries/website/javascript/index.md (hunk 1 above)
Comment (silesky) [reply src/connections/sources/catalog/libraries/website/javascript/index.md:667]: Oh no, I meant that `addSourceMiddleware` has two examples nested: "enrichment" and "validation"

File: src/connections/sources/catalog/libraries/website/javascript/index.md (hunk 5 above)
Comment (silesky) [reply src/connections/sources/catalog/libraries/website/javascript/index.md:668]: Yes

File: src/connections/sources/catalog/libraries/website/javascript/index.md (hunk 6 above)
Comment (silesky) [reply src/connections/sources/catalog/libraries/website/javascript/index.md:679]: Yes

File: src/connections/sources/catalog/libraries/website/javascript/index.md (hunk 7 above)
Comment (silesky) [reply src/connections/sources/catalog/libraries/website/javascript/index.md:690]: No -- I wanted to nest stuff underneath it =S But maybe H4

### Plugins  (for advanced use cases)
#### Plugin Categories, etc

**Comment by stayseesong** [comment]
@silesky sorry, I'm going to close this PR out. I accidentally did something funky with git but I remade the PR: #5705

**Pull request closed without merging**